src/library/columncache.cpp

```diff
@@ -181,7 +181,7 @@ void ColumnCache::slotSetKeySortOrder(double notation) {
     // A custom COLLATE function was tested, but using CASE ... WHEN was found to be faster
     // see GitHub PR#649
     // https://github.com/mixxxdj/mixxx/pull/649#discussion_r34863809
-    QString keySortSQL("CASE key_id ");
+    QString keySortSQL("CASE %1_id WHEN NULL THEN 0 ");
     for (int i = 0; i <= 24; ++i) {
             keySortSQL.append(QString("WHEN %1 THEN %2 ")
                 .arg(sortOrder[i]).arg(i));
```

Comment:
.arg().arg() is one of the QT anti pattern, because it parses the format string twice.  
This should be replaces by
.arg(sortOrder[i], QString::number(i));